src/spirc.rs

```diff
@@ -0,0 +1,251 @@
+
+use protobuf::{self, Message};
+
+use util;
+use session::{Config, Session};
+use util::SpotifyId;
+use util::version::version_string;
+use player::{Player, PlayerCommand};
+use mercury::{MercuryRequest, MercuryMethod};
+
+use librespot_protocol as protocol;
+use librespot_protocol::spirc::PlayStatus;
+
+pub struct SpircManager<'s> {
+    player: &'s Player<'s>,
+    session: &'s Session,
+
+    username: String,
+    state_update_id: i64,
+    seq_nr: u32,
+
+    name: String,
+    ident: String,
+    device_type: u8,
+    can_play: bool,
+
+    repeat: bool,
+    shuffle: bool,
+    volume: u16,
+
+    is_active: bool,
+    became_active_at: i64,
+
+    last_command_ident: String,
+    last_command_msgid: u32,
+
+    track: Option<SpotifyId>
+}
+
+impl <'s> SpircManager<'s> {
+    pub fn new(session: &'s Session, player: &'s Player<'s>, username: String, name: String) -> SpircManager<'s> {
```

Comment:
I was messing around a bit with the lifetime parameters here, did they end up correct? The struct is of type `SpircManager<'s>` so it should return one? And the session and player arguments?